### PR TITLE
Map manager branch

### LINKED ISSUE #8: [Feature]: MapManager subsystem
### Details of Feature

The subsystem "MapManager" as documented in the system design document & object design document

### PR DESCRIPTION
**Pull request by Jesse592**
Fixes #8 

The system creates the the middle man between the ViewModel and all the other systems
Summary of changes:
- Retreiving all routes from the database
- Calculcating a routefrom the database using Open Route Service
- Subscribing to the Location service to update the viewmodel
- Calculate the visited points from a route based on incoming location
- Notifying the ViewModel when a geofence istriggered
- Setting and updating the Geofence based on the selected route
- Many small extra's

### CODE REVIEW

File: app/src/main/java/nl/ags/picum/UI/MainActivity.java [hunk 1]
```diff
@@ -62,66 +74,39 @@ protected void onCreate(Bundle savedInstanceState) {
...
                 if (timeRequested < 2) {
-                    requestPermission(new String[]{permissions[i]});
+                    permissionstorequest.add(permissions[i]);
                 } else {
                     showPermissionDialog();
                 }
 
         }
         timeRequested++;
+        if (permissionstorequest.size() < 1)
```
Comment (JustRostin) [inline app/src/main/java/nl/ags/picum/UI/MainActivity.java]:
```suggestion
        if (permissionstorequest.size() >= 1)
```

File: app/src/main/java/nl/ags/picum/UI/MapActivity.java [hunk 2]
```diff
@@ -1,23 +1,91 @@
...
     }
 
     public void onStartRouteButtonClick(View view){
         ((Button)view).setVisibility(View.INVISIBLE);
         //TODO add function to start route
     }
+
+    public void initializeMap(){
+        mMap.setTileSource(TileSourceFactory.MAPNIK);
+        mMapController.setZoom(20.0);
```
Comment (JustRostin) [inline app/src/main/java/nl/ags/picum/UI/MapActivity.java]:
```suggestion
        mMapController.setZoom(20.1);
```

File: app/src/main/java/nl/ags/picum/dataStorage/managing/AppDatabaseManager.java [hunk 3]
```diff
@@ -10,6 +10,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import kotlin.NotImplementedError;
```
Comment (JustRostin) [inline app/src/main/java/nl/ags/picum/dataStorage/managing/AppDatabaseManager.java]: @Jesse592 ???? Kotlin??

**Review by JustRostin — COMMENTED** [review]
2 changes commiten (doe ik zelf)
1 vraag even antwoorden

File: app/src/main/java/nl/ags/picum/dataStorage/managing/AppDatabaseManager.java (hunk 3 above)
Comment (Jesse592) [reply app/src/main/java/nl/ags/picum/dataStorage/managing/AppDatabaseManager.java]: Not sure why this is in the database class, however the OkHttp dependency is from Kotling so seeing some kotlin imports is normal (Should only be in the RouteCalculator class)

Comment (Jesse592) [reply app/src/main/java/nl/ags/picum/dataStorage/managing/AppDatabaseManager.java]: @JustRostin Now fixed

**Review by JustRostin — APPROVED** [review]

**Pull request merged**